examples/bandwidthconstraints/src/index.js

```diff
@@ -0,0 +1,146 @@
+'use strict';
+
+var Prism = require('prismjs');
+var Video = require('twilio-video');
+var getRoomCredentials = require('../../util/getroomcredentials');
+var getSnippet = require('../../util/getsnippet');
+var helpers = require('./helpers');
+var waveform = require('../../util/waveform');
+var connectWithBandwidthConstraints = helpers.connectWithBandwidthConstraints;
+var updateBandwidthConstraints = helpers.updateBandwidthConstraints;
+
+var connectOrDisconnect = document.querySelector('input#connectordisconnect');
+var audioBitrateSelector = document.querySelector('select#maxaudiobitrate');
+var audioPreview = document.querySelector('audio#audiopreview');
+var videoBitrateSelector = document.querySelector('select#maxvideobitrate');
+var videoPreview = document.querySelector('video#videopreview');
+var waveformContainer = document.querySelector('div#audiowaveform');
+var roomName = null;
+var room = null;
+
+/**
+ * Attach the AudioTrack to the HTMLAudioElement and start the Waveform.
+ */
+function attachAudioTrack(track, audioElement) {
+  track.attach(audioElement);
+  waveform.setStream(audioPreview.srcObject);
+  var canvas = waveformContainer.querySelector('canvas');
+  if (!canvas) {
+    waveformContainer.appendChild(waveform.element);
+  }
+}
```

Comment:
We need to start factoring this stuff out, otherwise it will get difficult to maintain, IMO.

Reply:
@markandrus I've already moved `Waveform` to the `utils` folder.

Reply:
How about our "attach"/"detach" helpers? Can these be made common?

Reply:
@markandrus They're not entirely common. In this case, we're also attaching the Track to the Waveform module, whereas in other examples we are not. And I also don't want too much of the app logic to be modularized as it affects the flow of reading the code.